### PR TITLE
EVG-16139: Run test stats aggregations on test stats collection created by Presto

### PR DESCRIPTION
**Pull request by julianedwards**
JIRA: https://jira.mongodb.org/browse/EVG-16139

This is an intermediary step to ensure that the Presto data is correctly aggregating test results data and loading it to a collection in the Cedar DB (eventually this collection will be `historical_test_data` once we do the BSON to Parquet cutover) in a nightly ETL job.

This will not be merged right away, first we will create the MARS ETL job and then test in staging.

### CODE REVIEW

File: model/historical_test_data.go [hunk 1]
```diff
@@ -302,10 +307,33 @@ func GetHistoricalTestData(ctx context.Context, env cedar.Environment, filter Hi
...
 	}
 	if err = cursor.All(ctx, &data); err != nil {
-		return nil, errors.Wrap(err, "problem unmarshaling aggregated test data")
+		return nil, errors.Wrap(err, "unmarshaling aggregated test data")
+	}
+
+	// TODO (16140): Remove this comparison logic once we do the BSON to
+	// Parquet cutover.
+	var prestoData []AggregatedHistoricalTestData
+	cursor, err = env.GetDB().Collection(prestoHistoricalTestDataCollection).Aggregate(ctx, pipeline)
```
Comment (Kimchelly) [inline model/historical_test_data.go:319]: How does the data get into the `prestoHistoricalTestDataCollection` in the first place? Does it come from the MARS job?

File: model/historical_test_data.go [hunk 2]
```diff
@@ -302,10 +307,33 @@ func GetHistoricalTestData(ctx context.Context, env cedar.Environment, filter Hi
...
+	if err != nil {
+		grip.Warning(message.Fields{
+			"message": errors.Wrap(err, "aggregating Presto historical test data").Error(),
+			"filter":  filter,
+		})
+	} else {
+		if err = cursor.All(ctx, &prestoData); err != nil {
+			grip.Warning(message.Fields{
```
Comment (Kimchelly) [inline model/historical_test_data.go:327]:
```go
grip.Warning(message.WrapError(errors.Wrap(err, "..."), message.Fields{
    ...
}))
```

File: model/historical_test_data.go [hunk 3]
```diff
@@ -302,10 +307,33 @@ func GetHistoricalTestData(ctx context.Context, env cedar.Environment, filter Hi
...
-		return nil, errors.Wrap(err, "problem unmarshaling aggregated test data")
+		return nil, errors.Wrap(err, "unmarshaling aggregated test data")
+	}
+
+	// TODO (16140): Remove this comparison logic once we do the BSON to
+	// Parquet cutover.
+	var prestoData []AggregatedHistoricalTestData
+	cursor, err = env.GetDB().Collection(prestoHistoricalTestDataCollection).Aggregate(ctx, pipeline)
+	if err != nil {
+		grip.Warning(message.Fields{
```
Comment (Kimchelly) [inline model/historical_test_data.go:321]:
```go
grip.Warning(message.WrapError(errors.Wrap(err, "..."), message.Fields{
    ...
}))
```

**Review by Kimchelly — CHANGES_REQUESTED** [review]

File: model/historical_test_data.go (hunk 1 above)
Comment (julianedwards) [reply model/historical_test_data.go:319]: Yes! Ryan and I are going to write the MARS job Monday to do the nightly ETL of data.

**Review by Kimchelly — APPROVED** [review]

**Comment by julianedwards** [comment]
We have decided to not push this to production since it requires a backfill of all the test stats data that would be redundant—we already have the aggregated data in the `historical_test_data` collection—and cause unnecessary operational overhead. The presto test stats aggregation has been vetted in staging. Instead, we plan on writing the presto test stats aggregation data directly to the `historical_test_data` collection, first in staging then in prod, as part of the cutover in [EVG-16140](https://jira.mongodb.org/browse/EVG-16140).

**Pull request closed without merging**